src/abstractdataframe/selection.jl

```diff
@@ -770,6 +773,15 @@ function _add_multicol_res(res::DataFrameRow, newdf::DataFrame, df::AbstractData
     _insert_row_multicolumn(newdf, df, allow_resizing_newdf, colnames, res)
 end
 
+function _add_multicol_res(res::Tables.AbstractRow, newdf::DataFrame, df::AbstractDataFrame,
```

Comment:
While we're at it, maybe we should make `DataFrameRow <: AbstractRow`? That would avoid duplicating a few methods and simplifying type unions.

Reply:
I was thinking about it.

First (for future readers) `DataFrameRow` fully supports `Tables.AbstractRow` interface, so this is just an issue of code design.

Pros of doing subtyping:
* Less code.
* It is more clear which functionalities are on more "abstract level".

Cons of doing subtyping:
* Most of methods for `DataFrameRow` and `Tables.AbstractRow` are the same. However, not all of them. Some methods are different, because `DataFrameRow` has a richer functionality than `Tables.AbstractRow`. The challenge is that keeping `DataFrameRow` and `Tables.AbstractRow` separate makes it easier (at least for me) in the future to find all places in the source code where `DataFrameRow` is used. I know this is not a super strong reason but with the size of the code that we have I often end up doing updates of code by running "find in all files" of a certain code pattern (as otherwise it is easy to forget about some place when some functionality is used).

(for a reference: I started implementing this change and noticed that it would affect more code than only this PR and after this change)

So - we could add it, but it also has some practical downsides. What do you think, given this, we should do?

Reply:
I don't have a strong opinion. Given that Tables.jl uses duck typing anyway it's not super important to have `DataFrameRow <: AbstractRow`, and indeed nobody has requested it.

Reply:
Yes - duck typing is my main reason for sticking with what we have.